src/Service/RoleControlManager.php

```diff
@@ -48,16 +48,22 @@ public function getExtraFieldKey(string $name) {
   /**
    * {@inheritdoc}
    */
-  public function getUserFormMode(UserInterface $user) {
-    // TODO implement user form mode negotiation,
-    // which determine which role to use as main for user and fetch it settings.
-    return '';
+  public function getUserAccountFormMode(UserInterface $user) {
+    /** @var Drupal\user\Entity\RoleInterface $role */
+    $role = $this->getUserPriorityRole($user);
+    $third_party_settings = $role->getThirdPartySettings('role');
+    if (!$third_party_settings) {
+      return;
```

Comment:
Replace with `return NULL` this will be clear what it does.

Reply:
I think when we have `default` form mode we should return `NULL` and do nothing.